packages/react-dom/src/client/ReactDOM.js

```diff
@@ -186,17 +186,15 @@ type Batch = FiberRootBatch & {
 type Root = {
   render(children: ReactNodeList, callback: ?() => mixed): Work,
   unmount(callback: ?() => mixed): Work,
-  legacy_renderSubtreeIntoContainer(
-    parentComponent: ?React$Component<any, any>,
-    children: ReactNodeList,
-    callback: ?() => mixed,
-  ): Work,
-  createBatch(): Batch,
 
   _internalRoot: FiberRoot,
 };
 
-function ReactBatch(root: ReactRoot) {
+type RootWithBatchingAPI = Root & {
```

Comment:
Can we call this "ReactRoot", and instead call the other one "ReactSyncRoot"? These are just internal but these names tend to leak into official types in Flow/TypeScript.